### PR TITLE
MAINT Adding contiguous tests

### LINKED ISSUE #444: garbage error message resulting from type checks
image

"Type of A is not the Type of B" ? What is that supposed to mean for a user who just tried to call scattering on a `DoubleTensor`. We need a better error message here. Probably one stating that the input "probably needs to be of dtype float"

### LINKED ISSUE #529: Check for contiguous instead of casting in frontend
Is it possible to rather add a test, than `contiguous`. There will be a common test shared across backends in the PR `simplifications`.

_Originally posted by @edouardoyallon in https://github.com/kymatio/kymatio/pull/527_

This goes for other dimensionalities as well.

### PR DESCRIPTION
**Pull request by edouardoyallon**
Closes #529 
Closes #444

### CODE REVIEW

File: kymatio/scattering1d/frontend/tensorflow_frontend.py [hunk 1]
```diff
@@ -85,9 +85,6 @@ def scattering(self, x):
 
         return S
 
-    @tf.Module.with_name_scope
-    def __call__(self, x):
-        return self.scattering(x)
```
Comment (MuawizChaudhary) [inline kymatio/scattering1d/frontend/tensorflow_frontend.py]: Why?

File: kymatio/backend/numpy_backend.py [hunk 2]
```diff
@@ -1,5 +1,11 @@
 import numpy as np
 
+def input_checks(x):
+    if x is None:
+        raise TypeError('The input should be not empty')
+
+    if not x.flags['C_CONTIGUOUS'] or x is None:
+        raise RuntimeError('The input must be contiguous.')
```
Comment (MuawizChaudhary) [inline kymatio/backend/numpy_backend.py]: Not needed, me thinks. I've never encountered an error related to contiguity in NumPy ever. Torch, yes.

Comment (MuawizChaudhary) [reply kymatio/backend/numpy_backend.py]: Additionally we already checked if the input was empty previously.

File: kymatio/frontend/torch_frontend.py [hunk 3]
```diff
@@ -17,4 +18,7 @@ def scattering(self, x):
     def forward(self, x):
         """ This function provides the standard calling interface for PyTorch
         modules."""
+
+        input_checks(x)
+
```
Comment (MuawizChaudhary) [inline kymatio/frontend/torch_frontend.py]: We sometimes redefine the forward function in torch in some `*D` scatterings, to do size checks and such. This means we will not be doing several input_checks.

File: kymatio/scattering2d/frontend/tensorflow_frontend.py [hunk 4]
```diff
@@ -40,9 +40,5 @@ def scattering(self, input):
             S = tf.reshape(S, batch_shape + scattering_shape)
             return S
 
-    @tf.Module.with_name_scope
-    def __call__(self, x):
-        return self.scattering(x)
```
Comment (MuawizChaudhary) [inline kymatio/scattering2d/frontend/tensorflow_frontend.py]: Why

File: kymatio/backend/numpy_backend.py (hunk 2 above)
Comment (janden) [reply kymatio/backend/numpy_backend.py]: Agreed. No need for this in NP.

File: kymatio/backend/torch_backend.py [hunk 5]
```diff
@@ -3,6 +3,13 @@
 
 BACKEND_NAME = 'torch'
 
+def input_checks(x):
+    if x is None:
+        raise TypeError('The input should be not empty')
```
Comment (janden) [inline kymatio/backend/torch_backend.py]: Period.

Also, why not just make sure it's a `torch.Tensor`? Just checking for `None` is quite restricted.

File: kymatio/backend/torch_backend.py [hunk 6]
```diff
@@ -3,6 +3,13 @@
...
+
+    if not x.is_contiguous():
+        raise RuntimeError('The input must be contiguous')
```
Comment (janden) [inline kymatio/backend/torch_backend.py]: Period.

File: kymatio/frontend/torch_frontend.py (hunk 3 above)
Comment (janden) [reply kymatio/frontend/torch_frontend.py]: We do? We should only be redefining `scattering`, if I recall correctly.

File: kymatio/backend/numpy_backend.py [hunk 7]
```diff
@@ -1,5 +1,11 @@
 import numpy as np
 
+def input_checks(x):
+    if x is None:
+        raise TypeError('The input should be not empty')
```
Comment (janden) [inline kymatio/backend/numpy_backend.py]: Period.

Why not just make sure it's an `np.ndarray`?

File: kymatio/scattering1d/frontend/tensorflow_frontend.py (hunk 1 above)
Comment (janden) [reply kymatio/scattering1d/frontend/tensorflow_frontend.py]: It's defined in the main TF frontend.

File: kymatio/scattering2d/frontend/tensorflow_frontend.py (hunk 4 above)
Comment (janden) [reply kymatio/scattering2d/frontend/tensorflow_frontend.py]: It's defined in the main TF frontend.

File: kymatio/scattering3d/frontend/numpy_frontend.py [hunk 8]
```diff
@@ -54,31 +53,21 @@ def build(self):
         ScatteringBase3D.create_filters(self)
 
     def scattering(self, input_array):
-        methods = ['integral']
```
Comment (janden) [inline kymatio/scattering3d/frontend/numpy_frontend.py]: Wow this is weird. Thanks for fixing.

File: kymatio/backend/torch_backend.py (hunk 5 above)
Comment (janden) [reply kymatio/backend/torch_backend.py]: @edouardoyallon

**Pull request merged**